src/modules/EventModule.cpp

```diff
@@ -0,0 +1,463 @@
+	{
+		QStringList args = message.content().split(' ');
+
+		client.getGuildMember(channel.guildId(), message.author().id())
+			.then([=](const Discord::GuildMember& member)
+			{
+				bool found = false;
+
+				for (auto& allowedRole : m_EventWhitelist[channel.guildId()])
+				{
+					if (member.roles().contains(allowedRole))
+					{
+						found = true;
+						break;
+					}
+				}
+
+				if (!found)
+				{
+					UmikoBot::Instance().createMessage(channel.id(), "**You don't have permissions to end events.**");
+					return;
+				}
+				auto& config = getEventServerData(channel.guildId());
+				CurrencyModule* currencyModule = static_cast<CurrencyModule*>(UmikoBot::Instance().GetModuleByName("currency"));
+				auto& currencyConfig = currencyModule->getServerData(channel.guildId());
+
+				GuildSetting* setting = &GuildSettings::GetGuildSetting(channel.guildId());
+				if (args.size() != 1)
+				{
+					UmikoBot::Instance().createMessage(message.channelId(), "**Wrong Usage of Command!** ");
+					return;
+				}
+				if (config.isEventRunning == false)
+				{
+					UmikoBot::Instance().createMessage(message.channelId(), "**No event is running, so you can't end an event!**\n");
+					return;
+				}
+				if (config.eventTimer->isActive())
+				{
+					if (config.eventHighRiskHighRewardRunning)
+					{
+						Discord::Embed embed;
+						embed.setColor(15158332);
+						embed.setTitle("**HighRiskHighReward** event has **ended**!");
+
+						config.isEventRunning = false;
+						config.eventHighRiskHighRewardRunning = false;
+						config.eventTimer->stop();
+						currencyConfig.stealSuccessChance += highRiskRewardStealDecrease;
+						UmikoBot::Instance().createMessage(message.channelId(), embed);
+						return;
+					}
+					if (config.eventLowRiskLowRewardRunning)
+					{
+						Discord::Embed embed;
+						embed.setColor(15158332);
+						embed.setTitle("**LowRiskLowReward** event has **ended**!");
+
+						config.isEventRunning = false;
+						config.eventLowRiskLowRewardRunning = false;
+						config.eventTimer->stop();
+						currencyConfig.stealSuccessChance -= lowRiskRewardStealIncrease;
+						UmikoBot::Instance().createMessage(message.channelId(), embed);
+					}
+				}
+			});
+	});
+}
+
+void EventModule::OnMessage(Discord::Client& client, const Discord::Message& message)
+{
+	CurrencyModule* currencyModule = static_cast<CurrencyModule*>(UmikoBot::Instance().GetModuleByName("currency"));
+	client.getChannel(message.channelId()).then(
+		[this, message, &client, currencyModule](const Discord::Channel& channel)
+		{
+			if (channel.guildId() != 0 && !message.author().bot())
+			{
+				auto guildId = channel.guildId();
+				auto& config = getEventServerData(guildId);
+				auto& currencyConfig = currencyModule->getServerData(channel.guildId());
+
+				if (config.restored == true)
+				{
+					return;
+				}
+
+				if (config.isEventRunning == false)
+				{
+					config.restored = true;
+					return;
+				}
+				if (config.restored == false)
```

Comment:
You can simplify a lot of these `if`s.

If you've already checked for one condition, you (usually) don't need to check it again.

Reply:
Sure!